<!DOCTYPE html>
<html lang="en"><!--
 __  __                __                                     __
/\ \/\ \              /\ \             __                    /\ \
\ \ \_\ \   __  __    \_\ \      __   /\_\      __       ___ \ \ \/'\
 \ \  _  \ /\ \/\ \   /'_` \   /'__`\ \/\ \   /'__`\    /'___\\ \ , <
  \ \ \ \ \\ \ \_\ \ /\ \L\ \ /\  __/  \ \ \ /\ \L\.\_ /\ \__/ \ \ \\`\
   \ \_\ \_\\/`____ \\ \___,_\\ \____\ _\ \ \\ \__/.\_\\ \____\ \ \_\ \_\
    \/_/\/_/ `/___/> \\/__,_ / \/____//\ \_\ \\/__/\/_/ \/____/  \/_/\/_/
                /\___/                \ \____/
                \/__/                  \/___/

Powered by Hydejack v6.6.1 <https://qwtel.com/hydejack/>
--><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="format-detection" content="telephone=no"><meta http-equiv="x-ua-compatible" content="ie=edge"><title>BGGN-213, Lecture 18 (Part 2) &middot; BGGN 213</title><meta name="description" content="A hands-on introduction to the computer-based analysis of genomic and biomolecular data from the Division of Biological Sciences, UCSD. "><link rel="canonical" href="https://bioboot.github.io/bggn213_S19/class-material/lecture18_part2_example/"><link rel="alternate" type="application/atom+xml" title="BGGN 213 Feed" href="https://bioboot.github.io/bggn213_S19/feed.xml"><link rel="apple-touch-icon" href="/bggn213_S19/apple-touch-icon.png"><link id="_katexJS" rel="dns-prefetch" href="/bggn213_S19/assets/bower_components/katex/dist/katex.min.js"><link id="_katexCSS" rel="dns-prefetch" href="/bggn213_S19/assets/bower_components/katex/dist/katex.min.css"> <!--[if gt IE 8]><!----> <script> WebFontConfig = { google: { families: 'Roboto+Slab:700|Noto+Sans:400,400i,700,700i'.split('|') }, custom: { families: ['icomoon'], urls: ['/bggn213_S19/assets/icomoon/style.css'] } }; </script> <!--<![endif]--> <script> !function(n,e){function t(n,e){n.onload=function(){this.onerror=this.onload=null,e(null,n)},n.onerror=function(){this.onerror=this.onload=null,e(new Error("Failed to load "+this.src),n)}}function o(n,e){n.onreadystatechange=function(){"complete"!=this.readyState&&"loaded"!=this.readyState||(this.onreadystatechange=null,e(null,n))}}n._loaded=!1,n.loadJSDeferred=function(a,d){function r(){n._loaded=!0;var r=e.createElement("script");r.src=a,d&&(("onload"in r?t:o)(r,d),r.onload||t(r,d));var l=e.getElementsByTagName("script")[0];l.parentNode.insertBefore(r,l)}n._loaded?r():n.addEventListener?n.addEventListener("load",r,!1):n.attachEvent?n.attachEvent("onload",r):n.onload=r}}(window,document); !function(a){"use strict";var b=function(b,c,d){function e(a){return h.body?a():void setTimeout(function(){e(a)})}function f(){i.addEventListener&&i.removeEventListener("load",f),i.media=d||"all"}var g,h=a.document,i=h.createElement("link");if(c)g=c;else{var j=(h.body||h.getElementsByTagName("head")[0]).childNodes;g=j[j.length-1]}var k=h.styleSheets;i.rel="stylesheet",i.href=b,i.media="only x",e(function(){g.parentNode.insertBefore(i,c?g:g.nextSibling)});var l=function(a){for(var b=i.href,c=k.length;c--;)if(k[c].href===b)return a();setTimeout(function(){l(a)})};return i.addEventListener&&i.addEventListener("load",f),i.onloadcssdefined=l,l(f),i};"undefined"!=typeof exports?exports.loadCSS=b:a.loadCSS=b}("undefined"!=typeof global?global:this); !function(a){if(a.loadCSS){var b=loadCSS.relpreload={};if(b.support=function(){try{return a.document.createElement("link").relList.supports("preload")}catch(b){return!1}},b.poly=function(){for(var b=a.document.getElementsByTagName("link"),c=0;c<b.length;c++){var d=b[c];"preload"===d.rel&&"style"===d.getAttribute("as")&&(a.loadCSS(d.href,d,d.getAttribute("media")),d.rel=null)}},!b.support()){b.poly();var c=a.setInterval(b.poly,300);a.addEventListener&&a.addEventListener("load",function(){b.poly(),a.clearInterval(c)}),a.attachEvent&&a.attachEvent("onload",function(){a.clearInterval(c)})}}}(this); window.disablePushState = false; window.disableDrawer = false; </script> <!--[if lt IE 9]> <script src="/bggn213_S19/assets/bower_components/html5shiv/dist/html5shiv.min.js"></script> <![endif]--> <!--[if gt IE 8]><!----><style> article,aside,dialog,figcaption,figure,footer,header,hgroup,main,nav,section{display:block}mark{background:#FF0;color:#000}*{box-sizing:border-box}html,body{margin:0;padding:0}html{font-size:16px;line-height:1.75}body{color:#333;background-color:#fff;overflow-y:scroll}a{text-decoration:none}.lead{margin-left:-1rem;margin-right:-1rem}img,.img{display:block;max-width:100%;margin-bottom:1rem;border:none}img.lead,.img.lead{max-width:calc(100% + 2rem);width:calc(100% + 2rem)}h1,h2,h3,h4,h5,h6,.h1,.h2,.h3,.h4,.h5,.h6,.heading{font-weight:bold;text-rendering:optimizeLegibility}h1,h2,h3,h4,h5,h6,.h1,.h2,.h3,.h4,.h5,.h6{margin:3rem 0 1rem;line-height:1.6}h1,.h1{font-size:2rem;line-height:1.25}h2,.h2{font-size:1.5rem}h3,.h3{font-size:1.17em}p{margin-top:0;margin-bottom:1rem}p.lead{font-size:1.25rem;font-weight:300;padding:0 1rem}ul,ol,dl{margin-top:0;margin-bottom:1rem}ul,ol{padding-left:1.25rem}hr{position:relative;margin:1.5rem 0;border:0;border-top:1px solid #eee}.hr{padding-bottom:.5rem;border-bottom:1px solid #eee;margin-bottom:1.5rem}h4,h5,h6,.h4,.h5,.h6{margin-bottom:0.5rem;font-size:1rem}table{margin-bottom:1rem;width:100%;width:calc(100% + 2rem);margin-left:-1rem;border:1px solid #e5e5e5;border-collapse:collapse;border-spacing:0}td,th{padding:.25rem .5rem;border:1px solid #e5e5e5}td:first-child,th:first-child{padding-left:1rem}td:last-child,th:last-child{padding-right:1rem}thead+tbody,tbody+tbody,tfoot{border-top:3px double #e5e5e5}tbody tr:nth-child(odd) td,tbody tr:nth-child(odd) th{background-color:#f9f9f9}footer{margin-bottom:2rem}.page,.post{margin-bottom:3em}.page li+li,.post li+li{margin-top:.25rem}.page>header,.post>header{margin-bottom:2rem}.page-title,.post-title{margin-top:0}.post-date{display:block;margin-top:-0.5rem;margin-bottom:1rem;color:#9a9a9a}.related-posts{padding-left:0;list-style:none}.related-posts>li,.related-posts>li+li{margin-top:1rem}.related-posts>li>small,.related-posts>li+li>small{font-size:75%;color:#9a9a9a}.message{margin-bottom:1rem;padding:1rem;color:#787878;background-color:#f9f9f9;margin-left:-1rem;margin-right:-1rem}body,main{position:relative;overflow-x:hidden}@media screen{body::before{content:'';width:.5rem;background:#e5e5e5;position:absolute;left:0;top:0;bottom:0}}@media screen and (min-width: 40em){html{font-size:17px}}@media screen and (min-width: 54em){html{font-size:16px}}@media screen and (min-width: 72em){html{font-size:17px}}@media screen and (min-width: 125em){html{font-size:18px}}.sr-only{display:none}.clearfix,.sidebar-social::after,.clearafter::after{content:"";display:table;clear:both}a,.a{position:relative;padding-bottom:.15rem;border-bottom:1px solid}.img{overflow:hidden;background-color:#f9f9f9}.img>img{margin:0;width:100%;height:100%}.sixteen-nine{position:relative}.sixteen-nine::before{display:block;content:"";width:100%;padding-top:56.25%}.sixteen-nine>*{position:absolute;top:0;left:0;right:0;bottom:0}h1+hr,h2+hr,h3+hr,h4+hr,h5+hr,h6+hr{margin-top:0}.fade-in{animation-duration:500ms;animation-name:fade-in;animation-fill-mode:forwards}@keyframes fade-in{from{transform:translateY(-2rem);opacity:0}50%{transform:translateY(-2rem);opacity:0}to{transform:translateY(0);opacity:1}}.mb6{margin-bottom:6rem}.sidebar{color:rgba(255,255,255,0.75);text-align:left}.sidebar::before{content:"";position:absolute;z-index:2;top:0;left:0;bottom:0;right:0;background:linear-gradient(to bottom, rgba(32,32,32,0) 0%, rgba(32,32,32,0.5) 100%)}.sidebar a{color:#fff;border-bottom-color:rgba(255,255,255,0.2)}#_yDrawer{position:relative;padding:1rem 0}@media screen{#_yDrawer{padding:2rem 1rem;min-height:640px;min-height:100vh}}@media screen and (min-width: 54em){#_yDrawer{position:fixed;top:0;left:0;bottom:0;width:18rem;margin-left:0}}@media screen and (min-width: 97.5em){#_yDrawer{width:calc(50% - 28rem)}}@media screen{#_yDrawer.loaded{min-height:0;padding:0}}.sidebar-bg{position:absolute;top:0;right:0;bottom:0;left:0;background:#202020 center / cover}.sidebar-sticky{position:relative;z-index:3}@media screen{.sidebar-sticky{position:absolute;right:2rem;bottom:1rem;left:auto;max-width:14rem}}@media print{.sidebar-sticky{padding:0 2rem}}.sidebar-about>h1{color:#fff;font-size:2rem}.sidebar-nav>ul{list-style:none;padding-left:0;margin-bottom:.5rem}a.sidebar-nav-item{font-weight:bold;display:block;line-height:1.75;padding:.25rem 0;border-bottom:1px solid rgba(255,255,255,0.2)}@media screen{.y-drawer-scrim{z-index:2}.y-drawer-content{width:18rem;left:-17.5rem;z-index:3}}@media screen and (min-width: 97.5em){.y-drawer-content{width:calc(50% - 28rem)}}.sidebar-social{margin-bottom:.5rem}.sidebar-social>ul{list-style:none;padding-left:0;margin:0 -.25rem}.sidebar-social>ul>li{float:left;margin:0 .25rem}.sidebar-social>ul>li>a{display:inline-block;text-align:center;font-size:1.6rem;line-height:3rem;width:3.1249rem;height:4rem;padding:.5rem 0}.sidebar-social>ul li+li{margin-top:0}.fixed-top{position:fixed;top:0;left:0;width:100%;z-index:1}.navbar>.content{padding-top:0;padding-bottom:0;min-height:0;height:0}.menu{display:inline-block;padding:1.75rem 1.5rem;border-bottom:none;margin-left:-1.5rem;color:#9a9a9a !important}.menu::after{content:"\2630"}@media screen and (min-width: 54em){.menu{padding:1.25rem 1.5rem;position:absolute;left:-9999px}.menu:focus{position:static}}.animation-main{pointer-events:none}.loading{display:none}@media print{.menu{display:none}}.animation-main{opacity:0;will-change:opacity}.loading{position:absolute;top:0;right:0;padding:5.25rem 4.5rem;transform-origin:top right;transform:scale(0.33)}.content{position:relative;margin-left:auto;margin-right:auto;padding:5rem 1rem 12rem}@media screen{.content{max-width:38rem;padding-left:1.5rem;min-height:100vh}}@media screen and (min-width: 54em){.content{max-width:42rem;padding:4rem 1rem 12rem;margin-left:22rem;margin-right:4rem}}@media screen and (min-width: 72em){.content{margin-left:22rem;margin-right:4rem}}@media screen and (min-width: 92em){.content{max-width:48rem}}@media screen and (min-width: 97.5em){.content{margin:auto}}.me{float:right;width:6.5rem;height:6.5rem;margin-left:1rem;margin-bottom:.5rem;border-radius:100%;position:relative}@media screen and (min-width: 40em){.me{width:7rem;height:7rem}}@media screen and (min-width: 54em){.me{width:6.5rem;height:6.5rem}}@media screen and (min-width: 60em){.me{width:7rem;height:7rem}}main>footer{position:absolute;bottom:0;left:0;right:0;padding:0 1rem;color:#9a9a9a;font-size:smaller;text-align:center}main>footer>p{margin-bottom:0}html{font-family:Helvetica, Arial, sans-serif}h1,h2,h3,h4,h5,h6,.h1,.h2,.h3,.h4,.h5,.h6,.heading{font-family:Helvetica, Arial, sans-serif}</style><link id="_stylePreload" rel="preload" as="style" href="/bggn213_S19/assets/css/hydejack.css?v=6.6.1"> <script>document.getElementById('_stylePreload').onload=function(){this.rel='stylesheet'}</script><style id="_pageStyle"> .content a{color:#4f86aa;border-color:rgba(79,134,170,0.2)}.content a:hover{border-color:#4f86aa}:focus{outline-color:#4f86aa}::selection{color:#fff;background:#4f86aa}::-moz-selection{color:#fff;background:#4f86aa}</style><noscript><link rel="stylesheet" href="/bggn213_S19/assets/css/hydejack.css?v=6.6.1"><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab:700|Noto+Sans:400,400i,700,700i"><style> html { font-family: 'Noto Sans', Helvetica, Arial, sans-serif } h1, h2, h3, h4, h5, h6, .h1, .h2, .h3, .h4, .h5, .h6, .heading { font-family: 'Roboto Slab', Helvetica, Arial, sans-serif }</style><link rel="stylesheet" href="/bggn213_S19/assets/icomoon/style.css"> </noscript> <!--<![endif]--> <!--[if gt IE 8]><!----> <script async src="/bggn213_S19/assets/bower_components/webfontloader/webfontloader.js"></script> <!--<![endif]--><body><div class="navbar fixed-top"><div class="content"> <span class="sr-only">Jump to:</span> <a id="_menu" class="menu no-hover" href="#_title"> <span class="sr-only">Menu</span> </a></div></div><div id="_yPushState"><div class="fade-in"><main id="_main" class="content" role="main" data-color="#4f86aa" data-image="/bggn213_S19/assets/img/dna_BCK.jpg"><article class="page" role="article"><header><h1 class="page-title">BGGN-213, Lecture 18 (Part 2)</h1><div class="hr" style="padding-bottom:0"></div></header><h1 id="lecture18-investigating-cancer-genomics-datasets-part-2">Lecture18 Investigating cancer genomics datasets (Part 2)</h1><p class="message"><strong>BGGN213 Lecture 18:</strong><br /> Barry Grant &lt; <a href="http://thegrantlab.org">http://thegrantlab.org</a> &gt;<br /> Date: 2019-03-07 (16:30:10 PST on Thu, Mar 07)<p>This is a complement to the second hands-on session for <a href="https://bioboot.github.io/bimm213_W19/lectures/#18">lecture 18 of BGGN-213 W19</a>. You can find the Rmarkdown document that generated this page <a href="https://bioboot.github.io/bggn213_W18/class-material/lecture18_part2_example.Rmd">here</a>. In the following sections we walk through the analysis steps providing periodic output examples.<h2 id="identifing-sites-of-mutation">Identifing sites of mutation</h2><p>We start by <strong>1.</strong> reading the provided sequences (<a href="https://bioboot.github.io/bggn213_W19/class-material/lecture18_sequences.fa">lecture18_sequences.fa</a>) into R, then <strong>2.</strong> aligning, <strong>3.</strong> looking for sites of cancer specific mutation (i.e. differences between the two sequences), and finally <strong>4.</strong> outputing all 9-mer contaning subsequences encompasing these mutant sites.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">bio3d</span><span class="p">)</span><span class="w">
</span><span class="n">seqs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.fasta</span><span class="p">(</span><span class="s2">"~/Downloads/lecture18_sequences.fa"</span><span class="p">)</span><span class="w">
</span><span class="n">seqs</span><span class="w">
</span></code></pre></div></div><div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##              1        .         .         .         .         .         60 
## P53_wt       MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMDDLMLSPDDIEQWFTEDPGP
## P53_mutant   MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMLDLMLSPDDIEQWFTEDPGP
##              **************************************** ******************* 
##              1        .         .         .         .         .         60 
## 
##             61        .         .         .         .         .         120 
## P53_wt       DEAPRMPEAAPPVAPAPAAPTPAAPAPAPSWPLSSSVPSQKTYQGSYGFRLGFLHSGTAK
## P53_mutant   DEAPWMPEAAPPVAPAPAAPTPAAPAPAPSWPLSSSVPSQKTYQGSYGFRLGFLHSGTAK
##              **** ******************************************************* 
##             61        .         .         .         .         .         120 
## 
##            121        .         .         .         .         .         180 
## P53_wt       SVTCTYSPALNKMFCQLAKTCPVQLWVDSTPPPGTRVRAMAIYKQSQHMTEVVRRCPHHE
## P53_mutant   SVTCTYSPALNKMFCQLAKTCPVQLWVDSTPPPGTRVRAMAIYKQSQHMTEVVRRCPHHE
##              ************************************************************ 
##            121        .         .         .         .         .         180 
## 
##            181        .         .         .         .         .         240 
## P53_wt       RCSDSDGLAPPQHLIRVEGNLRVEYLDDRNTFRHSVVVPYEPPEVGSDCTTIHYNYMCNS
## P53_mutant   RCSDSDGLAPPQHLIRVEGNLRVEYLDDRNTFVHSVVVPYEPPEVGSDCTTIHYNYMCNS
##              ******************************** *************************** 
##            181        .         .         .         .         .         240 
## 
##            241        .         .         .         .         .         300 
## P53_wt       SCMGGMNRRPILTIITLEDSSGNLLGRNSFEVRVCACPGRDRRTEEENLRKKGEPHHELP
## P53_mutant   SCMGGMNRRPILTIITLEV-----------------------------------------
##              ******************                                           
##            241        .         .         .         .         .         300 
## 
##            301        .         .         .         .         .         360 
## P53_wt       PGSTKRALPNNTSSSPQPKKKPLDGEYFTLQIRGRERFEMFRELNEALELKDAQAGKEPG
## P53_mutant   ------------------------------------------------------------
##                                                                           
##            301        .         .         .         .         .         360 
## 
##            361        .         .         .  393 
## P53_wt       GSRAHSSHLKSKKGQSTSRHKKLMFKTEGPDSD
## P53_mutant   ---------------------------------
##                                                
##            361        .         .         .  393 
## 
## Call:
##   read.fasta(file = "~/Downloads/lecture16_sequences.fa")
## 
## Class:
##   fasta
## 
## Alignment dimensions:
##   2 sequence rows; 393 position columns (259 non-gap, 134 gap) 
## 
## + attr: id, ali, call
</code></pre></div></div><p>We can optionally align these sequences to make sure we have residue position correspondences correctly mapped between wt and mutant (incase of indels) with the following code. However, this appears to be unnecessary in this case as the provided sequences are already aligned.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#seqs &lt;- seqaln(seqs)</span><span class="w">
</span></code></pre></div></div><p>Next we calculate identity per equivalent (i.e. aligned) position and then use this information to find non identical sites that do not contain gaps (i.e. indels).<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Calculate positional identity scores</span><span class="w">
</span><span class="n">ide</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">conserv</span><span class="p">(</span><span class="n">seqs</span><span class="o">$</span><span class="n">ali</span><span class="p">,</span><span class="w"> </span><span class="n">method</span><span class="o">=</span><span class="s2">"identity"</span><span class="p">)</span><span class="w">
</span><span class="n">mutant.sites</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">which</span><span class="p">(</span><span class="n">ide</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> 

</span><span class="c1">## Exclude gap possitions from analysis</span><span class="w">
</span><span class="n">gaps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">gap.inspect</span><span class="p">(</span><span class="n">seqs</span><span class="p">)</span><span class="w">
</span><span class="n">mutant.sites</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mutant.sites</span><span class="p">[</span><span class="n">mutant.sites</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">gaps</span><span class="o">$</span><span class="n">f.inds</span><span class="p">]</span><span class="w">

</span><span class="n">mutant.sites</span><span class="w">
</span></code></pre></div></div><div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1]  41  65 213 259
</code></pre></div></div><p>We can use these indices in <code class="highlighter-rouge">mutant.sites</code> to extract subsequences as required for the hands-on session. First however we come up with suitable names for these subsequences based on the mutation. This will help us later to make sense and keep track of our results.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Make a "names" label for our output sequences (one per mutant)</span><span class="w">
</span><span class="n">mutant.names</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">paste0</span><span class="p">(</span><span class="n">seqs</span><span class="o">$</span><span class="n">ali</span><span class="p">[</span><span class="s2">"P53_wt"</span><span class="p">,</span><span class="n">mutant.sites</span><span class="p">],</span><span class="w">
                       </span><span class="n">mutant.sites</span><span class="p">,</span><span class="w">
                       </span><span class="n">seqs</span><span class="o">$</span><span class="n">ali</span><span class="p">[</span><span class="s2">"P53_mutant"</span><span class="p">,</span><span class="n">mutant.sites</span><span class="p">])</span><span class="w">

</span><span class="n">mutant.names</span><span class="w">
</span></code></pre></div></div><div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## [1] "D41L"  "R65W"  "R213V" "D259V"
</code></pre></div></div><p>Now lets extract all 9-mer mutant encompassing sequences for each mutant site. This is equivalent to finding the sequence region eight residues before and eight residues after our mutation sites and outputting this subsequence to a new FASTA file.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Sequence positions surounding each mutant site</span><span class="w">
</span><span class="n">start.position</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mutant.sites</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">8</span><span class="w">
</span><span class="n">end.position</span><span class="w"> </span><span class="o">&lt;-</span><span class="w">  </span><span class="n">mutant.sites</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">8</span><span class="w">

</span><span class="c1"># Blank matrix to store sub-sequences</span><span class="w">
</span><span class="n">store.seqs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="s2">"-"</span><span class="p">,</span><span class="w"> </span><span class="n">nrow</span><span class="o">=</span><span class="nf">length</span><span class="p">(</span><span class="n">mutant.sites</span><span class="p">),</span><span class="w"> </span><span class="n">ncol</span><span class="o">=</span><span class="m">17</span><span class="p">)</span><span class="w">
</span><span class="n">rownames</span><span class="p">(</span><span class="n">store.seqs</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">mutant.names</span><span class="w">

</span><span class="c1">## Extract each sub-sequence</span><span class="w">
</span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">mutant.sites</span><span class="p">))</span><span class="w"> </span><span class="p">{</span><span class="w">
  </span><span class="n">store.seqs</span><span class="p">[</span><span class="n">i</span><span class="p">,]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">seqs</span><span class="o">$</span><span class="n">ali</span><span class="p">[</span><span class="s2">"P53_mutant"</span><span class="p">,</span><span class="n">start.position</span><span class="p">[</span><span class="n">i</span><span class="p">]</span><span class="o">:</span><span class="n">end.position</span><span class="p">[</span><span class="n">i</span><span class="p">]]</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="n">store.seqs</span><span class="w">
</span></code></pre></div></div><div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##       [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12] [,13]
## D41L  "S"  "P"  "L"  "P"  "S"  "Q"  "A"  "M"  "L"  "D"   "L"   "M"   "L"  
## R65W  "D"  "P"  "G"  "P"  "D"  "E"  "A"  "P"  "W"  "M"   "P"   "E"   "A"  
## R213V "Y"  "L"  "D"  "D"  "R"  "N"  "T"  "F"  "V"  "H"   "S"   "V"   "V"  
## D259V "I"  "L"  "T"  "I"  "I"  "T"  "L"  "E"  "V"  "-"   "-"   "-"   "-"  
##       [,14] [,15] [,16] [,17]
## D41L  "S"   "P"   "D"   "D"  
## R65W  "A"   "P"   "P"   "V"  
## R213V "V"   "P"   "Y"   "E"  
## D259V "-"   "-"   "-"   "-"
</code></pre></div></div><p>Finally lets output all these sequences to a FASTA file for further analysis with the IEDB HLA binding prediction website <a href="http://tools.iedb.org/mhci/">http://tools.iedb.org/mhci/</a>.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## First blank out the gap positions </span><span class="w">
</span><span class="n">store.seqs</span><span class="p">[</span><span class="n">store.seqs</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"-"</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">""</span><span class="w">

</span><span class="c1">## Output a FASTA file for further analysis</span><span class="w">
</span><span class="n">write.fasta</span><span class="p">(</span><span class="n">seqs</span><span class="o">=</span><span class="n">store.seqs</span><span class="p">,</span><span class="w"> </span><span class="n">ids</span><span class="o">=</span><span class="n">mutant.names</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s2">"subsequences.fa"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h2 id="sidenote-input-sequence-setup">Sidenote: Input sequence setup</h2><p>For refernce only, here we use the UniProt KRas oncogene sequence (<a href="http://www.uniprot.org/uniprot/P01116">http://www.uniprot.org/uniprot/P01116</a>) as an example input and make 4 substations at random positions. Students would not need to do this as they will be provided with the output wild-type (<code class="highlighter-rouge">wt</code>) and mutant (<code class="highlighter-rouge">mutant</code>) containing FASTA format sequence file. We could also use p53 or any other protein for this hands-on session.<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">bio3d</span><span class="p">)</span><span class="w">

</span><span class="c1">## Read KRas oncogene sequence from UniProt</span><span class="w">
</span><span class="n">wt</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">get.seq</span><span class="p">(</span><span class="s2">"P01116"</span><span class="p">)</span><span class="w">

</span><span class="c1">## Here we make four mutants namely: G12V, Q22N, T74S and A130V</span><span class="w">
</span><span class="n">mutant</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">wt</span><span class="w">
</span><span class="n">mutant</span><span class="o">$</span><span class="n">ali</span><span class="p">[</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">12</span><span class="p">,</span><span class="m">22</span><span class="p">,</span><span class="m">74</span><span class="p">,</span><span class="m">130</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"V"</span><span class="p">,</span><span class="w"> </span><span class="s2">"N"</span><span class="p">,</span><span class="w"> </span><span class="s2">"S"</span><span class="p">,</span><span class="w"> </span><span class="s2">"V"</span><span class="p">)</span><span class="w">

</span><span class="n">write.fasta</span><span class="p">(</span><span class="w"> </span><span class="n">seqbind</span><span class="p">(</span><span class="n">wt</span><span class="p">,</span><span class="w"> </span><span class="n">mutant</span><span class="p">),</span><span class="w"> </span><span class="n">ids</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s2">"wt"</span><span class="p">,</span><span class="s2">"mutant"</span><span class="p">),</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s2">"kras-sequences.fa"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div><h2 id="session-info">Session Info</h2><div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sessionInfo</span><span class="p">()</span><span class="w">
</span></code></pre></div></div><div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Sierra 10.12.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] bio3d_2.3-3.9000
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.13    digest_0.6.12   rprojroot_1.2   R6_2.2.2       
##  [5] grid_3.4.1      backports_1.1.1 magrittr_1.5    evaluate_0.10.1
##  [9] httr_1.3.1      stringi_1.1.6   curl_3.0        rmarkdown_1.8  
## [13] tools_3.4.1     stringr_1.2.0   yaml_2.1.14     parallel_3.4.1 
## [17] compiler_3.4.1  htmltools_0.3.6 knitr_1.17
</code></pre></div></div></article><footer><hr/><p>Â© 2019 <a href="">Barry J. Grant</a>. All rights reserved. <code>A <a href="https://ucsd.edu">UCSD</a> Division of <a href="https://biology.ucsd.edu">Biological Sciences</a> Course</code></footer></main></div><div id="_yDrawer"><div id="_sidebar" class="sidebar"><div class="sidebar-bg" style="background-color:#4f86aa;background-image:url(/bggn213_S19/assets/img/dna_BCK.jpg)"></div><header class="sidebar-sticky" role="banner"> <img src="/bggn213_S19/assets/img/UCSanDiegoLogo-White.png" alt="UCSD Icon" width="400" height="76"><div class="sidebar-about"><h1><a id="_title" href="/bggn213_S19/">BGGN 213</a></h1><p>A hands-on introduction to the computer-based analysis of genomic and biomolecular data from the <a href="https://biology.ucsd.edu/">Division of Biological Sciences, UCSD</a>.</div><nav class="sidebar-nav heading" role="navigation"> <span class="sr-only">Navigation:</span><ul><li> <a class="sidebar-nav-item" href="/bggn213_S19/">Overview</a><li> <a class="sidebar-nav-item" href="/bggn213_S19/lectures/">Lectures</a><li> <a class="sidebar-nav-item" href="/bggn213_S19/setup/">Computer Setup</a><li> <a class="sidebar-nav-item" href="/bggn213_S19/goals/">Learning Goals</a><li> <a class="sidebar-nav-item" href="/bggn213_S19/midterm/">Assignments & Grading</a><li> <a class="sidebar-nav-item" href="/bggn213_S19/ethics/">Ethics Code</a></ul></nav><div class="sidebar-social"> <span class="sr-only">Social:</span><ul><li> <a href="https://twitter.com/barryjgrant" title="Twitter"> <span class="icon-twitter"></span> <span class="sr-only">Twitter</span> </a><li> <a href="https://github.com/bioboot" title="GitHub"> <span class="icon-github"></span> <span class="sr-only">GitHub</span> </a><li> <a href="mailto:<bjgrant@ucsd.edu>" title="Email"> <span class="icon-mail"></span> <span class="sr-only">Email</span> </a><li> <a href="https://bioboot.github.io/bggn213_S19/feed.xml" title="RSS"> <span class="icon-rss2"></span> <span class="sr-only">RSS</span> </a></ul></div></header></div></div></div><!--[if gt IE 9]><!----> <script>loadJSDeferred('/bggn213_S19/assets/js/hydejack.js?v=6.6.1');</script> <!--<![endif]-->
</html>
